docs/docs/components-and-props.md

```diff
@@ -0,0 +1,247 @@
+
+You can also use an [ES2015 class](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes) to define a component:
+
+```js
+class Welcome extends React.Component {
+  render() {
+    return <h1>Hello, {this.props.name}</h1>;
+  }
+}
+```
+
+The above two components are equivalent from React's point of view.
+
+Classes have some additional features that we will discuss in the next sections. Until then, we will use functional components for their conciseness.
+
+## Rendering a Component
+
+Previously, we only encountered React elements that represent DOM tags:
+
+```js
+const element = <div />;
+```
+
+However, elements can also represent user-defined components:
+
+```js
+const element = <Welcome name="Sara" />;
+```
+
+When React sees an element representing a user-defined component, it passes JSX attributes to this component as a single object. We call this object "props".
+
+For example, this code renders "Hello Sara" on the page:
+
+```js{1,5}
+function Welcome(props) {
+  return <h1>Hello, {props.name}</h1>;
+}
+
+const element = <Welcome name="Sara" />;
+const container = document.getElementById('root');
+ReactDOM.render(element, container);
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/YGYmEG?editors=0010)
+
+Let's recap what happens in this example:
+
+1. We call `ReactDOM.render(<Welcome name="Sara" />, container)`.
+3. React calls the `Welcome` component with `{name: 'Sara'}` as the props.
+4. Our `Welcome` component returns `<h1>Hello, Sara</h1>` element as the result.
+5. React DOM efficiently updates the DOM to match `<h1>Hello, Sara</h1>`.
+
+>**Caveat:**
+>
+>Always start component names with a capital letter. For example, `<div />` element represents a DOM tag, but `<Welcome />` represents a component. Also make sure your components are in scope when you are using them.
+
+## Composing Components
+
+Components can refer to other components in their output. This lets us use the same component abstraction for any level of detail. A button, a form, a dialog, a screen: in React apps, all those are commonly expressed as components.
+
+For example, we can create an `App` component that renders `Welcome` many times:
+
+```js{8-10}
+function Welcome(props) {
+  return <h1>Hello, {props.name}</h1>;
+}
+
+function App() {
+  return (
+    <div>
+      <Welcome name="Sara" />
+      <Welcome name="Cahal" />
+      <Welcome name="Edite" />
+    </div>
+  );
+}
+
+const container = document.getElementById('root');
+ReactDOM.render(<App />, container);
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/KgQKPr?editors=0010)
+
+Typically new React apps have a single `App` component as a root. However when you integrate an existing app with React, you might start bottom-up with a small component like `Button` and gradually convert more views to React components.
+
+>**Caveat:**
+>
+>Components must return a single root element. This is why we added a `<div>` to contain all the `<Welcome />` elements.
```

Comment:
I'm worried people will think "single element" = "has no children" so wanted to make it extra clear we're talking about roots.

I didn't close the tag because it has children. (Not sure it makes a lot of sense but the opposite is just as annoying to me.) Feel free to change this if you feel strongly about it.